packages/agent/src/agent/http/index.ts

```diff
@@ -833,14 +838,14 @@ export class HttpAgent implements Agent {
         throw new Error('Could not decode time from response');
       }
       const timeLookup = lookup_path(['time'], tree);
-      if (!timeLookup) {
+      if (timeLookup.status !== LookupStatus.Found) {
         throw new Error('Time was not found in the response or was not in its expected format.');
       }
 
-      if (!(timeLookup instanceof ArrayBuffer) && !ArrayBuffer.isView(timeLookup)) {
+      if (!(timeLookup.value instanceof ArrayBuffer) && !ArrayBuffer.isView(timeLookup)) {
```

Comment:
`timelookup.value` is either an `ArrayBuffer` or a  `HashTree`. So if we pass `timelookup.value` into `bufFromBufLike` when `timelookup.value` is a `HashTree`, it will pass the `Array.isArray` test in `bufFromBuflike` and be returned as an `ArrayBuffer`. And `decodeTime` may actually succeed, depending on the format of the `HashTree`, and have an incorrect timestamp returned.

It's more robust to exist early if we receive a malformed response in my opinion.